### ISSUE TITLE
ERROR in js/vendor.js from UglifyJs

### ISSUE THREAD

**Issue by jperelli** [quasarframework/quasar-cli#23]
`quasar dev` works without errors but when trying to build I get this error

```
$ quasar build


> quasar-app@0.0.1 build /home/jperelli/proteccionista-app
> node build/script.build.js

 WARNING!
 Do NOT use VueRouter's "history" mode if
 building for Cordova or Electron.

 Cleaned build artifacts.

 Building Quasar App with "mat" theme...

Build completed in 21.284s

Hash: df0b9ec307d771baac81
Version: webpack 2.2.1
Time: 21297ms
                                   Asset       Size  Chunks                    Chunk Names
            js/1.9171683b74816e9258f9.js    4.46 kB    1, 6  [emitted]         
fonts/MaterialIcons-Regular.012cf6a.woff    57.6 kB          [emitted]         
         fonts/Roboto-Light.37fbbba.woff    89.2 kB          [emitted]         
        fonts/Roboto-Medium.303ded6.woff    89.7 kB          [emitted]         
       fonts/Roboto-Regular.081b11e.woff    89.4 kB          [emitted]         
          fonts/Roboto-Thin.90d3804.woff    87.8 kB          [emitted]         
            js/0.436f2707e612d6e94430.js    5.06 kB    0, 6  [emitted]         
          fonts/Roboto-Bold.ad140ff.woff    89.2 kB          [emitted]         
            js/2.59c775d07c221c49f610.js    1.73 kB    2, 6  [emitted]         
            js/3.79d66cbac0c98d41bdec.js     2.7 kB    3, 6  [emitted]         
                            js/vendor.js    1.13 MB    4, 6  [emitted]  [big]  vendor
                               js/app.js    7.01 kB    5, 6  [emitted]         app
                          js/manifest.js    1.53 kB       6  [emitted]         manifest
app.3392693b69c210d69d3bacae149a4702.css     253 kB    5, 6  [emitted]  [big]  app
                              index.html  634 bytes          [emitted]         

ERROR in js/vendor.js from UglifyJs
SyntaxError: Name expected [js/vendor.js:37083,23]

 Build complete with "mat" theme in "/dist" folder.

 Built files are meant to be served over an HTTP server.
 Opening index.html over file:// won't work.
```

**Comment by rstoenescu** [comment]
Hi,

What NPM packages have you added?

**Comment by jperelli** [comment]
I think that only axios and vuex, not so sure about it.

```
  "dependencies": {
    "axios": "^0.15.3",
    "babel-runtime": "^6.0.0",
    "fastclick": "^1.0.6",
    "material-design-icons": "^3.0.1",
    "moment": "^2.15.0",
    "quasar-framework": "^0.13.1",
    "roboto-fontface": "^0.7.0",
    "vue": "2.1.10",
    "vue-router": "^2.0.0",
    "vuex": "^2.1.2"
  },
  "devDependencies": {
    "autoprefixer": "^6.4.0",
    "babel-core": "^6.0.0",
    "babel-eslint": "^7.0.0",
    "babel-loader": "^6.0.0",
    "babel-plugin-transform-runtime": "^6.0.0",
    "babel-preset-es2015": "^6.0.0",
    "babel-preset-stage-2": "^6.0.0",
    "colors": "^1.1.2",
    "connect-history-api-fallback": "^1.1.0",
    "css-loader": "^0.26.0",
    "eslint": "^3.0.1",
    "eslint-config-standard": "^6.2.0",
    "eslint-friendly-formatter": "^2.0.5",
    "eslint-loader": "^1.3.0",
    "eslint-plugin-html": "^1.3.0",
    "eslint-plugin-promise": "^3.3.0",
    "eslint-plugin-standard": "^2.0.0",
    "eventsource-polyfill": "^0.9.6",
    "express": "^4.13.3",
    "extract-text-webpack-plugin": "^2.0.0-beta.4",
    "file-loader": "^0.9.0",
    "friendly-errors-webpack-plugin": "^1.1.3",
    "html-webpack-plugin": "^2.8.1",
    "http-proxy-middleware": "^0.17.0",
    "json-loader": "^0.5.4",
    "opn": "^4.0.2",
    "postcss-loader": "^1.0.0",
    "progress-bar-webpack-plugin": "^1.9.0",
    "shelljs": "^0.7.0",
    "stylus": "^0.54.5",
    "stylus-loader": "^2.1.1",
    "url-loader": "^0.5.7",
    "vue-loader": "^10.0.0",
    "vue-style-loader": "^1.0.0",
    "vue-template-compiler": "2.1.10",
    "webpack": "^2.2.0",
    "webpack-dev-middleware": "^1.8.4",
    "webpack-hot-middleware": "^2.13.2",
    "webpack-merge": "^2.0.0"
  }
```

```
$ npm ls --depth=0
quasar-app@0.0.1 /home/jperelli/proteccionista-app
├── autoprefixer@6.7.3
├── axios@0.15.3
├── babel-core@6.23.1
├── babel-eslint@7.1.1
├── babel-loader@6.3.2
├── babel-plugin-transform-runtime@6.23.0
├── babel-preset-es2015@6.22.0
├── babel-preset-stage-2@6.22.0
├── babel-runtime@6.23.0
├── colors@1.1.2
├── connect-history-api-fallback@1.3.0
├── css-loader@0.26.1
├── eslint@3.15.0
├── eslint-config-standard@6.2.1
├── eslint-friendly-formatter@2.0.7
├── eslint-loader@1.6.1
├── eslint-plugin-html@1.7.0
├── eslint-plugin-promise@3.4.2
├── eslint-plugin-standard@2.0.1
├── eventsource-polyfill@0.9.6
├── express@4.14.1
├── extract-text-webpack-plugin@2.0.0-rc.3
├── fastclick@1.0.6
├── file-loader@0.9.0
├── friendly-errors-webpack-plugin@1.3.1
├── html-webpack-plugin@2.28.0
├── http-proxy-middleware@0.17.3
├── json-loader@0.5.4
├── material-design-icons@3.0.1
├── moment@2.17.1
├── opn@4.0.2
├── postcss-loader@1.3.1
├── progress-bar-webpack-plugin@1.9.3
├── quasar-framework@0.13.1
├── roboto-fontface@0.7.0
├── shelljs@0.7.6
├── stylus@0.54.5
├── stylus-loader@2.5.0
├── url-loader@0.5.7
├── vue@2.1.10
├── vue-loader@10.3.0
├── vue-router@2.2.1
├── vue-style-loader@1.0.0
├── vue-template-compiler@2.1.10
├── vuex@2.1.2
├── webpack@2.2.1
├── webpack-dev-middleware@1.10.1
├── webpack-hot-middleware@2.17.0
└── webpack-merge@2.6.1
```

**Comment by jperelli** [comment]
For what I'm seing in other places about this error, it might be something related to ES6 and babel that has to be transpiled before uglify?

**Comment by rstoenescu** [comment]
Can you please share your repo? Just want to investigate directly. Otherwise it's really hard to pinpoint the problem. Or can you look into js/vendor.js and copy paste the code around that line that causes error pls? `js/vendor.js:37083,23`. 10 lines before and after would be good.
Thanks!

**Comment by jperelli** [comment]
The repo is here https://github.com/jperelli/proteccionista-app

js/vendor.js with context:
```
"use strict";
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__util__ = __webpack_require__(217);
/* harmony export (immutable) */ __webpack_exports__["a"] = createLogger;
// Credits: borrowed code from fcomb/redux-logger



function createLogger ({         ## <--- line 37083
  collapsed = true,
  transformer = state => state,
  mutationTransformer = mut => mut
} = {}) {
  return store => {
    let prevState = __webpack_require__.i(__WEBPACK_IMPORTED_MODULE_0__util__["a" /* deepCopy */])(store.state)

    store.subscribe((mutation, state) => {
      if (typeof console === 'undefined') {
        return
      }
      const nextState = __webpack_require__.i(__WEBPACK_IMPORTED_MODULE_0__util__["a" /* deepCopy */])(state)
      const time = new Date()
      const formattedTime = ` @ ${pad(time.getHours(), 2)}:${pad(time.getMinutes(), 2)}:${pad(time.getSeconds(), 2)}.${pad(time.getMilliseconds(), 3)}`
      const formattedMutation = mutationTransformer(mutation)
      const message = `mutation ${mutation.type}${formattedTime}`
      const startMessage = collapsed
        ? console.groupCollapsed
        : console.group

      // render
      try {
        startMessage.call(console, message)
      } catch (e) {
        console.log(message)
      }

      console.log('%c prev state', 'color: #9E9E9E; font-weight: bold', transformer(prevState))
      console.log('%c mutation', 'color: #03A9F4; font-weight: bold', formattedMutation)
      console.log('%c next state', 'color: #4CAF50; font-weight: bold', transformer(nextState))

      try {
        console.groupEnd()
      } catch (e) {
        console.log('—— log end ——')
      }

      prevState = nextState
    })
  }
}
```

**Comment by rstoenescu** [comment]
From the looks of it, some package distributable is using ES6 code, which should not be the case. You got two options.. actually, I'd go with both:

1. Search node_modules for "Credits: borrowed code from fcomb/redux-logger" to see what package it is. And then log them a bug that they should not create distributable with ES6 code.

2. Add an exception for this package to your `/build/webpack.base.conf.js`. Edit:
```js
{
        test: /\.js$/,
        loader: 'babel-loader',
        include: projectRoot,
        exclude: /(node_modules|external)/   // <<<<<<<<<
},
```

**Comment by jperelli** [comment]
Nice, I found it

```
$ grep -nR "Credits: borrowed code from fcomb/redux-logger" node_modules/
node_modules/vuex/src/plugins/logger.js:1:// Credits: borrowed code from fcomb/redux-logger
node_modules/vuex/dist/logger.js:61:// Credits: borrowed code from fcomb/redux-logger
```

The problem was that I was importing the one in `src` folder (see https://github.com/jperelli/proteccionista-app/blob/f57d4b59b668deef41fa7e3de3eff5ee37867675/src/store/index.js#L3). That was just plain wrong.

And then, after all, this logger shouldn't go to production also, so I removed it.

But if I imported from `dist` folder (i.e. `import createLogger from 'vuex/dist/logger'`, as it [states in documentation](https://vuex.vuejs.org/en/plugins.html#built-in-logger-plugin)), it works. I think the problem was that there is an arrow function that uglify doesn't get.

Thanks for the help, works just right now!
